lib/shared/config-manager/src/index.ts

```diff
@@ -63,28 +84,121 @@ export class EnvironmentConfigManager {
                 this.logger.debug('DevCycle initial config loaded')
             })
             .finally(() => {
-                if (this.disablePolling) {
-                    return
-                }
-                this.intervalTimeout = this.setInterval(async () => {
-                    try {
-                        await this._fetchConfig()
-                    } catch (ex) {
-                        this.logger.error((ex as Error).message)
-                    }
-                }, this.pollingIntervalMS)
+                this.startPolling(this.configPollingIntervalMS)
+                this.startSSE()
             })
     }
+
+    private startSSE(): void {
+        if (!this.enableRealtimeUpdates) return
+
+        if (!this.configSSE) {
+            this.logger.warn('No SSE configuration found')
+            return
+        }
+        if (this.sseConnection) {
+            return
+        }
+
+        const url = new URL(
+            this.configSSE.path,
+            this.configSSE.hostname,
+        ).toString()
+        this.logger.debug(`Starting SSE connection to ${url}`)
+
+        this.sseConnection = new SSEConnection(url, this.logger, {
+            onMessage: this.onSSEMessage.bind(this),
+            onOpen: () => {
+                this.logger.debug('SSE connection opened')
+                // Set config polling interval to 10 minutes
+                this.startPolling(this.sseConfigPollingIntervalMS)
+            },
+            onConnectionError: () => {
+                this.logger.debug('SSE connection error, switching to polling')
+                // reset polling interval to default
+                this.startPolling(this.configPollingIntervalMS)
+                this.stopSSE()
```

Comment:
why does it need to stop and kill the SSE connection when theres an error? can't it just reconnect?

Reply:
my understanding of the SSE library is that it already re-connects automatically, and this get's called if it fails to re-connect. So in that case we should just close the SSE connection and go back to polling. But will double check that.

Reply:
it looks like it will try and reconnect every 1 sec: https://github.com/EventSource/eventsource/blob/db562b82fffb58b77cd6bb817a835db5c97a2398/lib/eventsource.js#L55

Reply:
and when does it call the error handler?

Reply:
check the `_emit('error', new Event('error')` messages in that code. I believe those are all the calls to `onerror`. Its not really clear to me if it ever not retries the connection. Maybe we need to close the connection based on the status codes?